### ISSUE TITLE
Malformed HTML when replicating metadata fields

### ISSUE THREAD

**Issue by nolanlawson** [mastodon/mastodon#8718]
If you compare the JSON coming from these two account API requests:

- https://mastodon.laurakalbag.com/api/v1/accounts/1
- https://toot.cafe/api/v1/accounts/66004

In the first you'll see the `fields` values contain well-formed HTML, but in the second one the HTML is malformed. For instance, notice the `&lt;` in this:

```js
"<a href=\"http://abookapart.com/products/accessibility-for-everyone\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">http://</span><span class=\"ellipsis\">abookapart.com/products/access</span><span class=\"invisible\">ibility-for-everyone&lt;/</span></a>"
```

mastodon.laurakalbag.com is running Mastodon v2.5.0, as is toot.cafe.

<details>
<summary>
Click here to see full JSON for <code>fields</code> values
</summary>

mastodon.laurakalbag.com version:

```json
{
    "fields": [
        {
            "name": "Ind.ie website",
            "value": "<a href=\"http://ind.ie\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">http://</span><span class=\"\">ind.ie</span><span class=\"invisible\"></span></a>"
        },
        {
            "name": "Better website",
            "value": "<a href=\"http://better.fyi\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">http://</span><span class=\"\">better.fyi</span><span class=\"invisible\"></span></a>"
        },
        {
            "name": "Buy my book at",
            "value": "<a href=\"http://abookapart.com/products/accessibility-for-everyone\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">http://</span><span class=\"ellipsis\">abookapart.com/products/access</span><span class=\"invisible\">ibility-for-everyone&lt;/</span></a>"
        },
        {
            "name": "Listen to my book at",
            "value": "<a href=\"http://www.audible.co.uk/pd/Accessibility-for-Everyone-Audiobook/B07G1SR1KQ?source_code=AUKFrDlWS02231890H6-BK-ACX0-123554&amp;ref=acx_bty_BK_ACX0_123554_rh_uk\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">http://www.</span></a>"
        }
    ]
}
```

toot.cafe version:

```json
{
    "fields": [
        {
            "name": "Ind.ie website",
            "value": "<a href=\"http://ind.ie\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">http://</span><span class=\"\">ind.ie</span><span class=\"invisible\"></span></a>"
        },
        {
            "name": "Better website",
            "value": "<a href=\"http://better.fyi\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">http://</span><span class=\"\">better.fyi</span><span class=\"invisible\"></span></a>"
        },
        {
            "name": "Buy my book at",
            "value": "<a href=\"http://abookapart.com/products/accessibility-for-everyone\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">http://</span><span class=\"ellipsis\">abookapart.com/products/access</span><span class=\"invisible\">ibility-for-everyone&lt;/</span></a>"
        },
        {
            "name": "Listen to my book at",
            "value": "<a href=\"http://www.audible.co.uk/pd/Accessibility-for-Everyone-Audiobook/B07G1SR1KQ?source_code=AUKFrDlWS02231890H6-BK-ACX0-123554&amp;ref=acx_bty_BK_ACX0_123554_rh_uk\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">http://www.</span></a>"
        }
    ]
}
```

</details>

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by ThibG** [comment]
This is because Mastodon truncates the fields to a length of at most 255 characters, and there is a discrepancy between local fields (internally unformatted) and remote ones (HTML-formatted).

**Comment by vmstan** [comment]
Is this still an issue?

**Comment by ClearlyClaire** [comment]
I think this has been addressed in #8868
